### ISSUE TITLE
NPE with AdoptOpenJDK 11 and SNI client

### ISSUE THREAD

**Issue by borkdude** [http-kit/http-kit#440]
I tried the SNI client on AdoptOpenJDK 11. This didn't work for me. When I use a different JVM, e.g. GraalVM JDK 11, it does work.

Repro:

``` 
borkdude@MBP2019 /tmp $ java -version
openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.3+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.3+7, mixed mode)
borkdude@MBP2019 /tmp $ clj -Sdeps '{:deps {http-kit/http-kit {:mvn/version "2.5.0-alpha2"}}}'
Clojure 1.10.1
user=> (require '[org.httpkit.client :as client])
nil
user=> (require '[org.httpkit.sni-client :as sni-client])
nil
user=> (alter-var-root #'org.httpkit.client/*default-client* (fn [_] sni-client/default-client))
#object[clojure.lang.Delay 0x7158daf2 {:status :pending, :val nil}]
user=> @(org.httpkit.client/get "https://www.google.com")
{:opts {:method :get, :url "https://www.google.com"}, :error #error {
 :cause nil
 :via
 [{:type java.lang.NullPointerException
   :message nil
   :at [sun.net.util.IPAddressUtil textToNumericFormatV4 "IPAddressUtil.java" 49]}]
 :trace
 [[sun.net.util.IPAddressUtil textToNumericFormatV4 "IPAddressUtil.java" 49]
  [sun.net.util.IPAddressUtil isIPv4LiteralAddress "IPAddressUtil.java" 241]
  [sun.security.util.HostnameChecker isIpAddress "HostnameChecker.java" 117]
  [sun.security.util.HostnameChecker match "HostnameChecker.java" 95]
  [sun.security.ssl.X509TrustManagerImpl checkIdentity "X509TrustManagerImpl.java" 459]
  [sun.security.ssl.X509TrustManagerImpl checkIdentity "X509TrustManagerImpl.java" 445]
  [sun.security.ssl.AbstractTrustManagerWrapper checkAdditionalTrust "SSLContextImpl.java" 1583]
  [sun.security.ssl.AbstractTrustManagerWrapper checkServerTrusted "SSLContextImpl.java" 1524]
  [sun.security.ssl.CertificateMessage$T12CertificateConsumer checkServerCerts "CertificateMessage.java" 620]
  [sun.security.ssl.CertificateMessage$T12CertificateConsumer onCertificate "CertificateMessage.java" 461]
  [sun.security.ssl.CertificateMessage$T12CertificateConsumer consume "CertificateMessage.java" 361]
  [sun.security.ssl.SSLHandshake consume "SSLHandshake.java" 392]
  [sun.security.ssl.HandshakeContext dispatch "HandshakeContext.java" 448]
  [sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction run "SSLEngineImpl.java" 1065]
  [sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction run "SSLEngineImpl.java" 1052]
  [java.security.AccessController doPrivileged "AccessController.java" -2]
  [sun.security.ssl.SSLEngineImpl$DelegatedTask run "SSLEngineImpl.java" 999]
  [org.httpkit.client.HttpsRequest doHandshake "HttpsRequest.java" 89]
  [org.httpkit.client.HttpClient doRead "HttpClient.java" 193]
  [org.httpkit.client.HttpClient run "HttpClient.java" 494]
  [java.lang.Thread run "Thread.java" 834]]}}
user=> ^D
```

**Comment by borkdude** [comment]
This could be related, not sure: https://bugs.openjdk.java.net/browse/JDK-8211339

**Comment by borkdude** [comment]
Seems to be fixed with 11.0.8!

``` 
borkdude@MBP2019 /tmp $ java -version
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.8+10, mixed mode)
borkdude@MBP2019 /tmp $ clj -Sdeps '{:deps {http-kit/http-kit {:mvn/version "2.5.0-alpha2"}}}'
Clojure 1.10.1
user=> (require '[org.httpkit.client :as client])
nil
user=> (require '[org.httpkit.sni-client :as sni-client])
nil
user=> (alter-var-root #'org.httpkit.client/*default-client* (fn [_] sni-client/default-client))
#object[clojure.lang.Delay 0x7158daf2 {:status :pending, :val nil}]
user=> (keys @(org.httpkit.client/get "https://www.google.com"))
(:opts :body :headers :status)
```

**Comment by ptaoussanis** [comment]
@borkdude Hi Michiel! Happy to hear you found a solution, and thanks for updating! Cheers :-)
